### PR TITLE
Update tests for prisons

### PR DESCRIPTION
**Pull request by danielatdigital**
It's possible that one prison name may actually be a subset of another
Check that the code will test for this

### CODE REVIEW

**Comment by ollietreend** [comment]
The logic looks good @danielatdigital, but what if the shorter prison name appears first in the list of prisons? I have a suspicion that if HMP Bourne was in the prisons list before HMP Bournemouth, the name matching could work slightly differently. It'd be worth testing that I think.

**Comment by danielatdigital** [comment]
@ollietreend I tried that first, and it works exactly the same way. Wasn't convinced it was worth having both versions of effectively the same test in the code

**Comment by ollietreend** [comment]
@danielatdigital I still think there's something we're missing. I've just looked over the `find_in_string` method again, as well as the `bad_data?` method of the Vacancy class.

What happens in this scenario:

Prison list only contains HMP Bourne

Vacancy title contains "HMP Bournemouth & HMP Bourne"

I don't think that would result in `bad_data?` returning `true`, but I think it should.

File: spec/wcn_scraper/prison_spec.rb [hunk 1]
```diff
@@ -142,8 +142,86 @@
...
+      end
+    end
+    context 'given a string containing two prisons, the substring site should only return once' do
+      subject(:prisons) { described_class.find_in_string('201706: Prison Officer - HMP Bourne & HMP Bournemouth') }
+
+      it 'returns an array of Prison instances' do
+        expect(prisons).to all(be_a(described_class))
+      end
+
+      it 'returns two results' do
```
Comment (ollietreend) [inline spec/wcn_scraper/prison_spec.rb]: This should read `it 'returns one result'`

Comment (danielatdigital) [reply spec/wcn_scraper/prison_spec.rb]: Done

**Pull request merged**